uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/calendar-ctrl.js

```diff
@@ -0,0 +1,283 @@
+        || status.name === 'approved'
+        || status.name === 'admin_approved') {
+        absenceType = _.find(this.absenceTypes, function (absenceType) {
+          return absenceType.id == leaveRequest.type_id;
+        });
+
+        //If Balance change is positive, mark as Accrued TOIL
+        if (leaveRequest.balance_change > 0) {
+          dateObj.UI.isAccruedTOIL = true;
+          return {
+            border: '1px solid ' + absenceType.color
+          };
+        }
+
+        return {
+          backgroundColor: absenceType.color,
+          borderColor: absenceType.color
+        };
+      }
+    };
+
+    /**
+     * Returns whether a date is of a specific type
+     * half_day_am or half_day_pm
+     *
+     * @param  {string} name
+     * @param  {object} leaveRequest
+     * @param  {string} date
+     *
+     * @return {boolean}
+     */
+    this._isDayType = function (name, leaveRequest, date) {
+      var dayType = this._dayTypes[name];
+
+      if (moment(date).isSame(leaveRequest.from_date)) {
+        return dayType.value == leaveRequest.from_date_type;
+      }
+
+      if (moment(date).isSame(leaveRequest.to_date)) {
+        return dayType.value == leaveRequest.to_date_type;
+      }
+    };
+
+    /**
+     * Returns whether a leaveRequest is pending approval
+     *
+     * @param  {object} leaveRequest
+     * @return {boolean}
+     */
+    this._isPendingApproval = function (leaveRequest) {
+      var status = this._leaveRequestStatuses[leaveRequest.status_id];
+
+      return status.name === 'waiting_approval';
+    };
+
+    /**
+     * Loads the absence periods
+     *
+     * @return {Promise}
+     */
+    this._loadAbsencePeriods = function () {
+      var self = this;
+
+      return AbsencePeriod.all()
+        .then(function (absencePeriods) {
+          self.absencePeriods = absencePeriods;
+          self.selectedPeriod = _.find(self.absencePeriods, function (period) {
+            return !!period.current;
+          });
+
+          self._fetchMonthsFromPeriod();
+        });
+    };
+
+    /**
+     * Loads the absence types
+     *
+     * @return {Promise}
+     */
+    this._loadAbsenceTypes = function () {
+      var self = this;
+
+      return AbsenceType.all()
+        .then(function (absenceTypes) {
+          self.absenceTypes = absenceTypes;
+        });
+    };
+
+    /**
+     * Loads the leave request day types
+     *
+     * @return {Promise}
+     */
+    this._loadDayTypes = function () {
+      var self = this;
+
+      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_day_type')
+        .then(function (dayTypesData) {
+          self._dayTypes = _.indexBy(dayTypesData, 'name');
+        });
+    };
+
+    /**
+     * Loads all the public holidays
+     *
+     * @return {Promise}
+     */
+    this._loadPublicHolidays = function () {
+      var self = this;
+
+      return PublicHoliday.all()
+        .then(function (publicHolidaysData) {
```

Comment:
not sure why this use of `Data` as a suffix. This is a list of public holidays, so just call it `publicHolidays`